pyoptsparse/pyNLPQLP/pyNLPQLP.py

```diff
@@ -30,9 +30,8 @@ def __init__(self, raiseError=True, options={}):
         category = "Local Optimizer"
         defOpts = self._getDefaultOptions()
         informs = self._getInforms()
-        if nlpqlp is None:
-            if raiseError:
-                raise Error("There was an error importing the compiled nlpqlp module")
+        if isinstance(nlpqlp, str) and raiseError:
```

Comment:
Related to my other comment, why check for strings and then throw the same error that we could throw in the `try_import_compiled_module_from_path` function?

Reply:
This is in the class init, so it's only executed when the class is instantiated.